app/main.py

```diff
@@ -1,8 +1,32 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_dict in people:
+        name = person_dict["name"]
+        age = person_dict["age"]
+        person = Person(name, age)
+        person_list.append(person)
+
+    for person_dict in people:
+        name = person_dict["name"]
+        spouse_name = person_dict.get("wife") or person_dict.get("husband")
+
+        if spouse_name:
+            person_instance = Person.people[name]
```

Comment:
Accessing `Person.people[name]` assumes that the person with the given name always exists in the dictionary. Consider adding error handling to manage cases where the name might not be found.